### PR TITLE
Implements identity match

### LINKED ISSUE #9: Identity Match isn't supported
Hey there, not sure if you're still working with this library or not. I've found it useful recently, thank you for what you've provided -- I was able to get Link up and running in minutes. For the Identity product, we decided to go with a Plaid beta product called [Identity Match](https://plaid.com/docs/api/products/identity/#identitymatch) which isn't yet supported here. 

It took a bit longer to get up and running since the response objects weren't present here, but it works fine. If you're interested in adopting the code, I can package it up and open a PR for it.

### PR DESCRIPTION
**Pull request by robacarp**
fixes #9 

👋 

I did my best to follow design patterns herein, but feel free to nitpick on style. I'll take no offense if you decide to merge it and rewrite it in your style, too! 😄 

I added a few typo-fix commits as well, if you're welcome to them.

In addition to the mock-http style testing, I've tested this by linking our app to the local copy of elixir-plaid and running the request against the sandbox API.

### CODE REVIEW

File: lib/plaid/identity.ex [hunk 1]
```diff
@@ -86,4 +86,75 @@ defmodule Plaid.Identity do
...
+            region: "NY",
+            postal_code: "10001",
+            country: "US"
+          }
+        }
+      }
+
+  ## Example
+
+      user_identity = =%{
```
Comment (mackshkatz) [inline lib/plaid/identity.ex:135]:
```suggestion
      user_identity = %{
```

**Comment by tylerwray** [comment]
Thank you very much for contributing! ❤️ 

I've merged this and will publish an updated version of the package here shortly.

**Comment by tylerwray** [comment]
Alright I've pushed up v1.2.0 with your changes! Check out the docs https://hexdocs.pm/elixir_plaid/Plaid.Identity.html#match/3

I did end up making some minor changes for consistency within the project. You can see that commit [here](https://github.com/tylerwray/elixir-plaid/commit/6e8d55ab41656435d330f709d97c40115388bb94).

Thanks again for your help!

**Comment by robacarp** [comment]
Awesome thank you @tylerwray 🎉 Those changes all make sense, and thank you for tidying up 😃 
I'm sorry you had to go through so much churn in your CI, and I appreciate it.

**Pull request merged**